components/gripper/softrobotics/gripper.go

```diff
@@ -32,24 +32,27 @@ type AttrConfig struct {
 }
 
 // Validate ensures all parts of the config are valid.
-func (cfg *AttrConfig) Validate(path string) error {
+func (cfg *AttrConfig) Validate(path string) ([]string, error) {
+	var deps []string
 	if cfg.Board == "" {
-		return utils.NewConfigValidationFieldRequiredError(path, "board")
+		return nil, utils.NewConfigValidationFieldRequiredError(path, "board")
 	}
 	if cfg.Open == "" {
-		return utils.NewConfigValidationFieldRequiredError(path, "open")
+		return nil, utils.NewConfigValidationFieldRequiredError(path, "open")
 	}
 	if cfg.Close == "" {
-		return utils.NewConfigValidationFieldRequiredError(path, "close")
+		return nil, utils.NewConfigValidationFieldRequiredError(path, "close")
 	}
 	if cfg.Power == "" {
-		return utils.NewConfigValidationFieldRequiredError(path, "power")
+		return nil, utils.NewConfigValidationFieldRequiredError(path, "power")
 	}
 
 	if cfg.AnalogReader != "psi" {
```

Comment:
I am not familiar with the softrobotic gripper, looks like its supposed to be a pin so make sense its required

Reply:
This is because of the ADC magic we do to make this gripper work with a pi. We can remove all these requirements and return warnings if we ever want to break this code.